### PR TITLE
Onboarding v2 dp

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves : https://github.com/brave/brave-browser/issues/9555

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Follow test plan on the issue. and follow the design spec as mention on the issue.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by mariospr** [comment]
I rebased today on top of the latest master and I can't build on Linux anymore because of this PR, more specifically because of this: https://github.com/brave/brave-core/commit/72df475de3ffd6931c2a3e2d43195e72be216abb#diff-e9394688e89210103ace82f5b1b746ba

It seems like that commit introduced a dependency on android-specific code from `components/brave_perf_predictor/browser/perf_predictor_tab_helper.cc`, which is also built on (non-android) Linux.

I looked into the changes in there and I suppose it should be possible to add some preprocessor conditions to make sure that those includes, and the code used from them, are not used in Linux, but @deeppandya will probably know better :slightly_smiling_face:

File: components/brave_perf_predictor/browser/perf_predictor_tab_helper.cc [hunk 1]
```diff
@@ -23,10 +25,10 @@ content::WebContents* GetWebContents(int render_process_id,
                                      int render_frame_id,
                                      int frame_tree_node_id) {
   content::WebContents* web_contents =
-      content::WebContents::FromFrameTreeNodeId(frame_tree_node_id);
```
Comment (iefremov) [inline components/brave_perf_predictor/browser/perf_predictor_tab_helper.cc:26]: @deeppandya would you mind restoring proper formatting as per Chromium code style? You can set up `clang-format` to automatically format code in your IDE

File: BUILD.gn [hunk 2]
```diff
@@ -108,6 +108,12 @@ if (is_android) {
       "$root_build_dir/brave_resources.pak",
       "$root_build_dir/brave_100_percent.pak",
       "//brave/LICENSE.html",
+      "//brave/android/java/org/chromium/chrome/browser/onboarding/animations/onboarding_ads.json",
+      "//brave/android/java/org/chromium/chrome/browser/onboarding/animations/privacy_protection.json",
+      "//brave/android/java/org/chromium/chrome/browser/onboarding/animations/save_data_and_battery.json",
+      "//brave/android/java/org/chromium/chrome/browser/onboarding/animations/website_loads_faster.json",
+      "//brave/android/java/org/chromium/chrome/browser/onboarding/animations/onboarding_ads_notification.json",
```
Comment (iefremov) [inline BUILD.gn:115]: we should keep all these lists sorted, `gn format` makes it easy to fix

**Comment by srirambv** [comment]
Verification passed on OnePlus 6T with Android 10 running 1.14.44 x64 nightly build

**Clean Install**
- Verified privacy stats onboarding is shown the moment browser is launched

First info card about `Privacy protection` highlights `Trackers & Ads blocked` | Second info card about `Save data and battery` highlights `Data Saved`  | Third info card about `Websites loads faster` highlights `Est. Time saved` 
-- | -- | --
image | image | image

- The fourth info card about privacy stats doesn't highlight the stats button on NTP. Logged #11325
- Verified clicking `Turn on privacy reports` on the first screen and following through brings up the privacy reports modal once onboarding is completed

Light theme | Dark Theme
-- | --
light-theme-onboarding | dark-theme-onboarding

- Verified swiping across the privacy onboarding brings up the fourth info card and clicking on `Learn more` starts the onboarding again

Light theme | Dark Theme
-- | --
light-theme-skip-onboarding | dark-theme-skip-onboarding

- Clicking on Individual stats on NTP doesn't bring up the respective info card. Logged #11319

**Search Engine discovery**
- Verified clicking on the URL bar brings up the SE onboarding 
- Verified changing default search engine via onboarding is retained
- Verified changing/retaining default search engine is set for both normal and private tabs

**Brave Shields**
- Verified visiting a page for the first time where nothing is blocked doesn't bring out the shields message
- Verified visiting a page for the first time shows shields onboarding message when the page load is done
- Verified clicking on the shields message opens the Shields panel 
- Onboarding message is not dismissed when a new tab is opened before page load completes #11318

**Rewards**
- Verified on clean install clicking on the rewards button brings up rewards onboarding as per spec
- Verified if not enabled, Rewards onboarding is shown upon launch after 4 days
- Clicking on sample ad notification doesn't load the landing page. Logged #11323

**Brave Desktop intro**
- Verified Brave Desktop intro card is shown after 12 days
- Verified intro card is shown on a private tab if the browser is launched via private tab on the home screen

Light | Dark
-- | --
image| image

- Logged #11326 for suggesting text colour change to not make it look like a clickable link 

**Upgrade Install**
- Verified stat button is shown on an upgrade on NTP
- Verified interaction hotspot is shown on the stats button on NTP
- Verified clicking it loads the privacy stats onboarding
- Logged #11324 for not showing the first info card in onboarding

---------------------------------
Verification passed on Samsung Tab A with Android 10 running 1.14.44 x64 nightly build

**Clean Install**
- Verified privacy stats onboarding is shown the moment browser is launched

First info card about `Privacy protection` highlights `Trackers & Ads blocked` | Second info card about `Save data and battery` highlights `Data Saved`  | Third info card about `Websites loads faster` highlights `Est. Time saved` 
-- | -- | --
image | image | image

- The fourth info card about privacy stats doesn't highlight the stats button on NTP. Logged #11325
- Verified clicking `Turn on privacy reports` on the first screen and following through brings up the privacy reports modal once onboarding is completed

Light theme | Dark Theme
-- | --
tab-light-theme-onboarding | tab-dark-theme-onboarding

- Verified swiping across the privacy onboarding brings up the fourth info card and clicking on `Learn more` starts the onboarding again

Light theme | Dark Theme
-- | --
tab-light-theme-skip-onboarding | tab-dark-theme-skip-onboarding

- Clicking on Individual stats on NTP doesn't bring up the respective info card. Logged #11319

**Search Engine discovery**
- Verified clicking on the URL bar brings up the SE onboarding 
- Verified changing default search engine via onboarding is retained
- Verified changing/retaining default search engine is set for both normal and private tabs

**Brave Shields**
- Verified visiting a page for the first time where nothing is blocked doesn't bring out the shields message
- Verified visiting a page for the first time shows shields onboarding message when the page load is done
- Verified clicking on the shields message opens the Shields panel 
- Onboarding message is not dismissed when a new tab is opened before page load completes #11318

**Rewards**
- Verified on clean install clicking on the rewards button brings up rewards onboarding as per spec
- Verified if not enabled, Rewards onboarding is shown upon launch after 4 days
- Clicking on sample ad notification doesn't load the landing page. Logged #11323

**Brave Desktop intro**
- Verified Brave Desktop intro card is shown after 12 days.
- Verified intro card is shown on a private tab if the browser is launched via private tab on the home screen

Light | Dark
-- | --
image | image

- Logged #11326 for suggesting text colour change to not make it look like a clickable link 

**Upgrade Install**
- Verified stat button is shown on an upgrade on NTP
- Verified interaction hotspot is shown on the stats button on NTP
- Verified clicking it loads the privacy stats onboarding
- Logged #11324 for not showing the first info card in onboarding

**Tab specific issues**
- Privacy report modal scale is pretty small on tablet - #11322
- SE onboarding on tablet isn't smooth - #11201 

### Note: Translations are not checked for any of the above

File: android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java [hunk 3]
```diff
@@ -284,6 +313,52 @@ public void didSelectTab(Tab tab, @TabSelectionType int type, int lastId) {
     };
   }
 
+  private void addSavedBandwidthToDb(long savings) {
```
Comment (bridiver) [inline android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java:316]: this is ridiculously overcomplicating things. The value is already stored in a pref that we can read directly in java. Storing this stuff in a database is completely unnecessary

**Pull request merged**